### ISSUE TITLE
[studio] Data grid density issue

### ISSUE THREAD

**Issue by oliviertassinari** [mui/toolpad#4242]
### Search keywords

-

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

The current row height for the different density settings looks odd. They were first defined with https://github.com/mui/mui-x/issues/567 but they don't seem to work super great.  It feels like everything is so big, for example, no https://github.com/mui/mui-public/pull/211, this is not compact, this is medium size.

A quick benchmark:

|                                                               | -1          | Default     | +1          |
|---------------------------------------------------------------|-------------|-------------|-------------|
| MUI X                                                         | 36px (x0.7) | 52px (spec) | 67px (x1.3) |
| [Elastic](https://eui.elastic.co/#/tabular-content/data-grid) | 23px        | 35px        | 39px        |
| Gmail                                                         | 28px        | 40px        |             |
| Airtable                                                      | 31px        | 55px        | 87px        |

I'm not super clear on either the issue is rooted in Toolpad. It could be the default values of MUI X that are too big, but they are following the Material Design spec, so hard to argue for a change.

At the end of the day, I suspect that it's a problem with Material UI. We should fix https://github.com/mui/material-ui/issues/41579, which Toolpad would then leverage.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by KenanYusuf** [comment]
> I'm not super clear on either the issue is rooted in Toolpad. It could be the default values of MUI X that are too big, but they are following the Material Design spec, so hard to argue for a change.

I agree that it is hard to argue a change for the defaults in MUI X, given they follow the spec. The density can look off based on row content. I would suggest adjusting the `rowHeight` value on the data grid in Toolpad to suit the use cases there.

> At the end of the day, I suspect that it's a problem with Material UI. We should fix https://github.com/mui/material-ui/issues/41579, which Toolpad could then leverage.

This is a really nice suggestion. We could have a global density setting that applies to all Material UI components (including data grid) which can be overwritten on a per-component basis.
